sa/sa.go

```diff
@@ -402,6 +414,46 @@ func (ssa *SQLStorageAuthority) countCertificatesByName(domain string, earliest,
 	return len(serialMap), nil
 }
 
+// countCertificatesByExactNames returns, for a single domain, the count of
+// certificates issued in the given time range for that domain. In contrast to
+// the countCertificatesByNames function subdomains are NOT considered.
+//
+// The highest count this function can return is 10,000. If there are more
+// certificates than that matching one of the provided domain names, it will return
+// TooManyCertificatesError.
+func (ssa *SQLStorageAuthority) countCertificatesByExactName(domain string, earliest, latest time.Time) (int, error) {
```

Comment:
Good point about the deployability concern. And, coming back to this again, I realize that this is still non-overlapping. Specifically, `reversedDomain` here will never be a public suffix. So for instance, this would count:

`WHERE reversedName = "io.dedyn.jsha" OR reversedName LIKE CONCAT("io.dedyn.jsha, ".%")`

However, for the exact-match case we would be looking for:

`WHERE reversedName = "io.dedyn"`

This is fairly subtle; could you add a comment explaining it?